### ISSUE TITLE
"Available offline" does not seem to work

### ISSUE THREAD

**Issue by sciurius** [owncloud/android#2241]
### Actual behaviour
When a folder is flagged as "Avalable offline", it is assumed to be updated automatically (within a reasonable amount of time) when a file in this folder changes on the server.

This doesn't seem to work for me on all of my devices:
Samsung Note 10.1 (GT-N8010) with LineageOS 14.1 (Android 7.1)
Samsung A3 (A320FL) with Oreo 8.0
Asus Zenpad Z300M with Android 7.0

As soon as the ownCloud app is opened on the device, synching takes place.

### Expected behaviour
Synching should take place regardless of opening the app.
 
### Steps to reproduce
1. Mark a folder on the device as "available offline". It will be updated.
2. Modify on a PC a file in this folder. It will be uploaded to the server.
3. Watch the device (e.g., via an ssh connection) and see that the file is not updated even after a long time (over one hour).
4. Open the ownCloud app, the file is updated immediately (provided there is a network connection).

### Environment data
Android version: See above.

Device model: See above.

Stock or customized system:
GT-N8010: Custom.
A320FL and Z300M: stock.

ownCloud app version:
2.7.0

ownCloud server version:
10.0.8

#### Web server error log
n/a
#### ownCloud log (data/owncloud.log)
n/a
Insert your ownCloud log here
n/a

**Comment by theScrabi** [comment]
> Synching should take place regardless of opening the app.

It will, however Google changed the behavior for code that runs in the background. It is only possible to update every 15 minutes. So the file will be updated every 15 minutes if you don't open the app.

In order to show the latest changes you did on the server we of course also check marked files if you open the app.

**Comment by sciurius** [comment]
I would love to acknowledge, but I can't.

This is what I tried.

On the A320FL (hostname smotty), run a ssh server (Simple SSHD).
On the PC:
```
$ date ; ssh smotty ls -l /storage/emulated/0/owncloud/xx@yy.zz/Safe
Fri 15 Jun 10:30:49 CEST 2018
total 104
-rw-rw---- 1 root everybody  1123 2018-06-14 13:00 pwsafe.cfg
-rw-rw---- 1 root everybody 99528 2018-06-13 17:14 pwsafe.psafe3
```
On the PC, make a small change to Safe/pwsafe.cfg:
```
$ ls -l Safe/pwsafe.cfg
-rw-r--r-- 1 jv jv 1127 Jun 15 10:32 Safe/pwsafe.cfg
```
Verify that it is uploaded to the server:
```
$ tail -4 owncloudsync.log
#=#=#=# Syncrun started 2018-06-15T08:32:29Z
#=#=#=#=# Propagation starts 2018-06-15T08:32:30Z (last step: 606 msec, total: 606 msec)
08:32:33||Safe/pwsafe.cfg|INST_SYNC|Up|1529051547|0a3b660f4a2eae6ab1bbddd4e707fc58|1127|00000540oc7snh0a7v3h|4||204|1123|1528969165||||
#=#=#=# Syncrun finished 2018-06-15T08:32:33Z (last step: 3441 msec, total: 4048 msec)
```
Then, repeatedly run the ssh command to see if the file changes on the A320FL:
```
$ date ; ssh smotty ls -l /storage/emulated/0/owncloud/xx@yy.zz/Safe
Fri 15 Jun 10:49:05 CEST 2018
total 104
-rw-rw---- 1 root everybody  1123 2018-06-14 13:00 pwsafe.cfg
-rw-rw---- 1 root everybody 99528 2018-06-13 17:14 pwsafe.psafe3
$ date ; ssh smotty ls -l /storage/emulated/0/owncloud/xx@yy.zz/Safe
Fri 15 Jun 10:54:36 CEST 2018
total 104
-rw-rw---- 1 root everybody  1123 2018-06-14 13:00 pwsafe.cfg
-rw-rw---- 1 root everybody 99528 2018-06-13 17:14 pwsafe.psafe3

```
Note that we're already significantly past the 15 minute mark.
I attached two screenshots from the A320FL showing that the folder was marked to be available offline, and that automatic synching is enabled for the account (in fact, for all accounts).

![photo_2018-06-15_11-11-15](https://user-images.githubusercontent.com/1118804/41460568-b8b34f68-707c-11e8-9263-19c6de311862.jpg)
![photo_2018-06-15_11-11-23](https://user-images.githubusercontent.com/1118804/41460570-b9005088-707c-11e8-912e-f3ce0ca983c5.jpg)

**Comment by theScrabi** [comment]
Hm have you checked if only the date label is wrong or if the files really haven't been synchronized? Maybe you can install the [beta version](https://f-droid.org/repo/com.owncloud.android_20600100.apk) of our app [here from fdroid](https://f-droid.org/en/packages/com.owncloud.android/). This version can collect logs which you can find in the settings. It might help us if you send us those logs. I will also take a look at this and check it :)

**Comment by theScrabi** [comment]
Related to this: https://github.com/owncloud/android/issues/2124#issuecomment-371083397

**Comment by sciurius** [comment]
Thanks for the detailed explanation.
May I assume that using Android jobs for this is already planned for a near future version?

**Comment by michaelstingl** [comment]
@davigonz Is there already an issue for the implementation of "Android jobs"/"job scheduler"? Priority and roadmap should be discussed there…

**Comment by davigonz** [comment]
> Is there already an issue for the implementation of "Android jobs"/"job scheduler"? Priority and roadmap should be discussed there

Sure, #2124 is related and @jesmrec and I were trying to reproduce any bad behaviour with instant uploads in latest versions of Android but no progress so far.

**Comment by davigonz** [comment]
Implementation of available offline with Android jobs is already merged and will be released in 2.9 so  the local files will be checked every 15 minutes and automatically uploaded if there's any change, even with the app closed.
